Kickstarter-iOS/Features/RootTabBar/RootTabBarViewController.swift

```diff
@@ -260,46 +260,70 @@ public final class RootTabBarViewController: UITabBarController, MessageBannerVi
       case let .search(index):
         _ = tabBarItem(atIndex: index) ?|> searchTabBarItemStyle
       case let .dashboard(index):
-        _ = tabBarItem(atIndex: index) ?|> dashboardTabBarItemStyle
+        let featureFlaggedTabBarItemStyle = self
+          .isDashboardViewControllerDisplayable() ? dashboardTabBarItemStyle :
+          profileTabBarItemStyle(isLoggedIn: data.isLoggedIn, isMember: data.isMember)
+        _ = tabBarItem(atIndex: index) ?|> featureFlaggedTabBarItemStyle
```

Comment:
This function is called as a result of `tabData(forUser` and it checks if that tab is available before styling it. So if we sent "Explore, Activity, Search, Dashboard and Profile" to this function - the view controllers may only be "Explore, Activity, Search and Profile". Which means that `Dashboard` case updates its styling to reflect "Profile" and the `Profile` case doesn't update anything - the tab doesn't exist. `self.tabBar.items` is called under the hood for `tabData(forUser` and reflects the view controllers - which are independently set by `setViewControllers` (whenever that happens).

Reply:
The only trade-off is that we lose the Profile user avatar, if they were a creator logged in with the dashboard hidden. I think its' worth it because this only affects logged in creators and not backers and we're removing creator login functionality anyway.